packages/components/inputs/rich-text-input/src/rich-text-input.tsx

```diff
@@ -0,0 +1,254 @@
+import pick from 'lodash/pick';
+import { filterDataAttributes, warning } from '@commercetools-uikit/utils';
+import {
+  richTextPlugins,
+  html,
+  isEmpty,
+} from '@commercetools-uikit/rich-text-utils';
+import renderEditor, { type TEditorProps } from './editor';
+
+type TBaseEvent = {
+  target: {
+    id?: string;
+    name?: string;
+  };
+};
+
+type TChangeEvent = {
+  target: TBaseEvent['target'] & {
+    value: string;
+  };
+};
+
+type TFocusEvent = TBaseEvent;
+
+type TEventHook<T = Event> = (
+  event: T,
+  editor: Editor,
+  next: () => unknown
+) => unknown;
+
+type TOnChangeParam = {
+  operations: unknown;
+  value: ReturnType<typeof html.deserialize>;
+};
+type TOnChangeFn = (change: TOnChangeParam) => unknown;
+
+export type TRichTextInputProps = {
+  /**
+   * Focus the control when it is mounted
+   */
+  isAutofocussed?: boolean;
+  /**
+   * Expands multiline text input initially
+   */
+  defaultExpandMultilineText?: TEditorProps['defaultExpandMultilineText'];
+  /**
+   * Indicates the input field has an error
+   */
+  hasError?: TEditorProps['hasError'];
+  /**
+   * Indicates the input field has warning
+   */
+  hasWarning?: TEditorProps['hasWarning'];
+  /**
+   * Used as the HTML `id` attribute.
+   */
+  id?: string;
+  /**
+   * Used as the HTML `name` attribute.
+   */
+  name?: string;
+  /**
+   * Placeholder value to show in the input field
+   */
+  placeholder: string;
+  /**
+   * Disables the rich text input
+   */
+  isDisabled?: TEditorProps['isDisabled'];
+  /**
+   * Indicates that the rich text input is displaying read-only content
+   */
+  isReadOnly?: TEditorProps['isReadOnly'];
+  /**
+   * Horizontal size limit of the input fields
+   */
+  horizontalConstraint?: TEditorProps['horizontalConstraint'];
+  /**
+   * Called with an event containing the new value. Required when input is not read only. Parent should pass it back as value.
+   */
+  onChange?: (event: TChangeEvent) => void;
+  /**
+   * Called when input is focused
+   */
+  onFocus?: (event: TFocusEvent) => void;
+  /**
+   * Called when input is blurred
+   */
+  onBlur?: (event: TFocusEvent) => void;
+  /**
+   * Value of the input component.
+   */
+  value?: string;
+  /**
+   * Indicates whether the expand icon should be visible
+   */
+  showExpandIcon: TEditorProps['showExpandIcon'];
+  /**
+   * Called when the `expand` button is clicked
+   */
+  onClickExpand?: TEditorProps['onClickExpand'];
+};
+
+class RichTextInput extends PureComponent<TRichTextInputProps> {
+  static defaultProps: Pick<
+    TRichTextInputProps,
+    | 'defaultExpandMultilineText'
+    | 'horizontalConstraint'
+    | 'placeholder'
+    | 'showExpandIcon'
+  > = {
+    defaultExpandMultilineText: false,
+    horizontalConstraint: 'scale',
+    placeholder: '',
+    showExpandIcon: false,
+  };
+
+  static displayName = 'RichTextInput';
+  static isEmpty = isEmpty;
+  static isTouched = (touched?: boolean) => Boolean(touched);
```

Comment:
Maybe we should need to allow both `boolean` and `boolean[]` types for the way this function is used from formik state.
I saw that pattern in other PRs lile [this one](https://github.com/commercetools/ui-kit/blob/main/packages/components/inputs/select-input/src/select-input.tsx#L497).